core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -147,7 +147,7 @@ protected static long guardedRamBytesUsed(Accountable a) {
      * Tries to extract a segment reader from the given index reader.
      * If no SegmentReader can be extracted an {@link IllegalStateException} is thrown.
      */
-    protected static SegmentReader segmentReader(LeafReader reader) {
+    public static SegmentReader segmentReader(LeafReader reader) {
```

Comment:
nit: If this utility method get more use-cases, I'd move it to a separate class, like maybe the "Lucene" class in that case?

Reply:
done, good point